### PR TITLE
Guardian Service Delete by Id function & Guardian Service Add function

### PR DESCRIPTION
**Pull request by viralpandya**
image

### CODE REVIEW

**Comment by viralpandya** [comment]
All the conflicts are resolved and those has both the branches. Add and Delete. Please check.

File: OtripleS.Web.Api.Tests.Unit/Services/GuardianServiceTests/GuardianServiceTests.Exceptions.Add.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    public partial class GuardianServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnAddWhenSqlExceptionOccursAndLogItAsync()
+        {
+            // given
+            DateTimeOffset dateTime = GetRandomDateTime();
+            Guardian randomGuardian = CreateRandomGuardian(dateTime);
+            Guardian inputGuardian = randomGuardian;
+            inputGuardian.UpdatedBy = inputGuardian.CreatedBy;
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Unit/Services/GuardianServiceTests/GuardianServiceTests.Exceptions.Add.cs:25]: What's the point of this line?

Comment (tajwal) [reply OtripleS.Web.Api.Tests.Unit/Services/GuardianServiceTests/GuardianServiceTests.Exceptions.Add.cs:25]: @viralpandya ,

Am waiting for Delete & post method of Guardian service to complete Guardiancontroller.

**Comment by hassanhabib** [comment]
@viralpandya please address conflicts and questions above.

File: OtripleS.Web.Api.Tests.Unit/Services/GuardianServiceTests/GuardianServiceTests.Exceptions.Add.cs (hunk 1 above)
Comment (viralpandya) [reply OtripleS.Web.Api.Tests.Unit/Services/GuardianServiceTests/GuardianServiceTests.Exceptions.Add.cs:25]: > What's the point of this line?

there is a testcase that if createdby and updated by are not same while adding new guardian, then it should fail on validation.

Comment (viralpandya) [reply OtripleS.Web.Api.Tests.Unit/Services/GuardianServiceTests/GuardianServiceTests.Exceptions.Add.cs:25]: > @viralpandya ,
> 
> Am waiting for Delete & post method of Guardian service to complete Guardiancontroller.

Updating it. Thank you for your patience!

**Comment by viralpandya** [comment]
resolved conflicts. @hassanhabib : please check

**Pull request merged**